code/clue.js

```diff
@@ -1,59 +1,284 @@
+  age: 68,
+  image: "assets/mustard.png",
+  occupation: "Retired captain",
+  favouriteWeapon: "dumbbelL",
+};
+
+//OBJECTS FOR ALL THE WEAPONS
 const rope = {
-  name: 'rope',
-  weight: 10
-}
+  name: "a rope",
+  weight: 10,
+  id: "ropE",
+};
+
+const knife = {
+  name: "a knife",
+  weight: 6,
+  id: "knifE",
+};
+
+const candlestick = {
+  name: "a candlestick",
+  weight: 12,
+  id: "candlesticK",
+};
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const dumbbell = {
+  name: "a dumbbell",
+  weight: 25,
+  id: "dumbbelL",
+};
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const poison = {
+  name: "a poison",
+  weight: 3,
+  id: "poisoN",
+};
 
+const axe = {
+  name: "an axe",
+  weight: 17,
+  id: "axE",
+};
+
+const bat = {
+  name: "a bat",
+  weight: 12,
+  id: "baT",
+};
+
+const trophy = {
+  name: "a trophy",
+  weight: 15,
+  id: "trophY",
+};
+
+const pistol = {
+  name: "a pistol",
+  weight: 8,
+  id: "pistoL",
+};
+
+// THE ROOMS ONLY HAVE A NAME SO NO NEED FOR OBJECTS
+
+//SUSPECTS, WEAPONS AND ROOMS GROUPED IN ARRAYS
+//array of objects
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  mrsWhite,
+  profPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
+];
+
+//array of strings
+const suspectsArray = [
+  "Mr Green",
+  "Professor Plum",
+  "Miss Scarlet",
+  "Mrs Peacock",
+  "Colonel Mustard",
+  "Mrs White",
+];
+
+//array of objects
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
 
-const weapons = []
+//array of strings
+const weaponsArray = [
+  "Rope",
+  "Knife",
+  "Candlestick",
+  "Dumbbell",
+  "Poison",
+  "Axe",
+  "Bat",
+  "Trophy",
+  "Pistol",
+];
 
-const rooms = []
+//array of strings
+const favouriteWeapon = [
+  "ropE",
+  "axE",
+  "poisoN",
+  "baT",
+  "candlesticK",
+  "dumbbelL",
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+//array of strings
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// function randomly selects one item from the array you pass in to the function
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// This function will be invoked when you click on the killer card.
+//an object that keeps the mystery
+const mysteryObject = {
+  killer: null,
+  weapon: null,
+  room: null,
```

Comment:
Great that you initialise your `mystery` object with `null` values, that's the cleanest approach from semantic point of view :)